src/cdk-experimental/listbox/listbox.ts

```diff
@@ -194,360 +158,382 @@ export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightab
     'role': 'listbox',
     'class': 'cdk-listbox',
     '[id]': 'id',
-    '(focus)': '_focusActiveOption()',
-    '(keydown)': '_keydown($event)',
-    '[attr.tabindex]': '_tabIndex',
+    '[attr.tabindex]': 'disabled ? null : 0',
     '[attr.aria-disabled]': 'disabled',
     '[attr.aria-multiselectable]': 'multiple',
     '[attr.aria-activedescendant]': '_getAriaActiveDescendant()',
     '[attr.aria-orientation]': 'orientation',
+    '(focus)': '_handleFocus()',
+    '(keydown)': '_handleKeydown($event)',
+    '(focusout)': '_handleFocusOut($event)',
   },
-  providers: [CDK_LISTBOX_VALUE_ACCESSOR],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CdkListbox),
+      multi: true,
+    },
+  ],
 })
-export class CdkListbox<T> implements AfterContentInit, OnDestroy, OnInit, ControlValueAccessor {
-  _listKeyManager: ActiveDescendantKeyManager<CdkOption<T>>;
-  _selectionModel: SelectionModel<CdkOption<T>>;
-  _tabIndex = 0;
-
-  /** `View -> model callback called when select has been touched` */
-  _onTouched: () => void = () => {};
-
-  /** `View -> model callback called when value changes` */
-  _onChange: (value: T) => void = () => {};
-
-  readonly optionSelectionChanges: Observable<OptionSelectionChangeEvent<T>> = defer(() => {
-    const options = this._options;
-
-    return options.changes.pipe(
-      startWith(options),
-      switchMap(() => merge(...options.map(option => option.selectionChange))),
-    );
-  }) as Observable<OptionSelectionChangeEvent<T>>;
-
-  private _disabled: boolean = false;
-  private _multiple: boolean = false;
-  private _useActiveDescendant: boolean = false;
-  private _autoFocus: boolean = true;
-  private _activeOption: CdkOption<T>;
-  private readonly _destroyed = new Subject<void>();
-
-  @ContentChildren(CdkOption, {descendants: true}) _options: QueryList<CdkOption<T>>;
-
-  @Output() readonly selectionChange = new EventEmitter<ListboxSelectionChangeEvent<T>>();
+export class CdkListbox<T = unknown> implements AfterContentInit, OnDestroy, ControlValueAccessor {
+  /** The id of the option's host element. */
+  @Input() id = `cdk-listbox-${nextListboxId++}`;
 
-  @Input() id = `cdk-listbox-${listboxId++}`;
+  // TODO(mmalerba): Should we normalize the order? Do we care about duplicates?
```

Comment:
Should we do any normalization of order/duplicates in the value? I could see duplicates being useful (see my comment above about the case where people omit a value). There's different options for how order the values:
- Lexicographical
- The order the options appear in
- The order the user clicked them (this one is coincidentally how it works now)

I could see cases where people might want different things e.g.:
- If you want it to behave like a set where you get the same value regardless of DOM order or selection order, lexicographical makes sense
- Maybe someone wants to allow the user to drag and drop the options, in this case oder in the DOM is probably what they want
- Someone might ask a user to "pick your top 3" in which case they might want the order the user clicked them in.

Reply:
I think that we started off with the selection order in `mat-select` and we eventually switched to DOM order, because that's what people were expecting. I think we should do the same here.

Reply:
+1 to DOM order

I think "order of selection" is the fuzziest / least expected.